task8-main/src/main/java/com/example/classloader/app/Demo.java

```diff
@@ -0,0 +1,82 @@
+
+        String pathName = getModulsLocation("location.properties");
+        List<File> jars = Arrays.asList(new File(pathName).listFiles());
+        URL[] urls = new URL[jars.size()];
+        for (int i = 0; i < jars.size(); i++) {
+            try {
+                urls[i] = jars.get(i).toURI().toURL();
+                System.out.println(urls[i]);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        URLClassLoader childClassLoader = new URLClassLoader(urls, ClassLoader.getSystemClassLoader());
+        Thread.currentThread().setContextClassLoader(childClassLoader);
+
+        LOGGER.info("Enter the operation name for calculation (Add/Multiply/Square Root): ");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        String operation = reader.readLine();
+
+        LOGGER.info("Enter the parameters to calculate the result of operation (press any character to exit): ");
+        Scanner in = new Scanner(System.in);
+
+        ArrayList<Double> paramList = new ArrayList<>();
+        while (in.hasNextDouble()) {
+            double param = in.nextDouble();
+            paramList.add(param);
+        }
+        LOGGER.info(paramList.toString());
+
+        double result = performCalculation(childClassLoader, operation, paramList);
+        LOGGER.info("The result is " + result);
+    }
+
+    private static double performCalculation(URLClassLoader childClassLoader, String operationName, ArrayList<Double> paramList) throws ClassNotFoundException, InvocationTargetException, InstantiationException, IllegalAccessException, NoSuchMethodException {
+
+        switch (operationName.toLowerCase()) {
+            case "add":
+                return getCalculationResult(childClassLoader, "com.example.classloader.implement.CalculateAdd", paramList);
+            case "multiply":
+                return getCalculationResult(childClassLoader, "com.example.classloader.implement.CalculateMultiply", paramList);
+            case "square root":
+                return getCalculationResult(childClassLoader, "com.example.classloader.implement.CalculateSquareRoot", paramList);
+            default:
+                LOGGER.warn("The operation name entered is not correct!");
+                return 0;
+        }
+    }
+
+    private static double getCalculationResult(URLClassLoader childClassLoader, String binaryClassName, ArrayList<Double> paramList) throws ClassNotFoundException, InvocationTargetException, InstantiationException, IllegalAccessException, NoSuchMethodException {
+        Class[] parameterType = new Class[]{ArrayList.class};
+        Class myClass = childClassLoader.loadClass(binaryClassName);
+        Constructor constructor = myClass.getConstructor(parameterType);
+        Object calculate = constructor.newInstance(paramList);
```

Comment:
not object, but your Calculable interface

and then - no find Method, but call it explicit